PROJECT_GUIDELINES.md

```diff
@@ -0,0 +1,228 @@
+    );
+  }
+
+  // ...
+}
+```
+
+This will create a unique id attribute like `id="calcite-example-51af-0941-54ae-22c14d441beb"`, which should have a VERY low collision change since guid() generates IDs with `window.crypto.getRandomValues`. If a user supplies an `id`, it will respect the users `id`.
+
+##### i18n
+
+Components should require as a few text translations as possible. In general, this allows users to supply text values via slots and attributes and handle translations within their apps.
+
+If your component involves formatting numbers or dates use the [`Intl` APIs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) for formating the display of numbers and dates in the component.
+
+To add RTL support to your components, use the internal `getElementDir` helper to apply the `dir` attribute to the component. That way, the `dir` attribute of the component will always match that of the document.
+
+```tsx
+import { Component, Host, Element, h} from "@stencil/core";
+import { getElementDir } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-component",
+  styleUrl: "calcite-component.scss",
+  shadow: true
+})
+export class CalciteComponent {
+  @Element() el: HTMLElement;
+
+  // ...
+
+  render() {
+    const dir = getElementDir(this.el);
+
+    return (
+      <Host dir={dir}>
+        <!-- The rest of your component -->
+      </Host>
+    );
+  }
+}
+```
+
+Direction specific CSS can be implemented with CSS variables:
+
+```scss
+:host {
+  --calcite-tabs-tab-margin-start: 1.25rem;
+  --calcite-tabs-tab-margin-end: 0;
+}
+
+:host([dir="rtl"]) {
+  --calcite-tabs-tab-margin-start: 0;
+  --calcite-tabs-tab-margin-end: 1.25rem;
+}
+```
+
+- https://medium.com/stencil-tricks/implementing-internationalisation-i18n-with-stencil-5e6559554117 and https://codesandbox.io/s/43pmx55vo9
+- https://github.com/ionic-team/ionic-stencil-conference-app/issues/69
+
+#### Testing
+
+Please see Stencil's doc for [end-to-end testing](https://stenciljs.com/docs/end-to-end-testing)!
+
+Components should have an automated test for any incoming features or bug fix. We utilize Travis CI to check our work, so PRs will run tests and now allow merging unless all the tests pass (we suggest running locally before making a PR to save time). We encourage writing expressive test cases and code that indicates intent. Use comments sparingly when the aforementioned can't be fully achieved. Keep it clean!
+
+#### Browser support
+
+Our components are best used in most modern browsers including Chrome, Firefox and Safari.
+
+## Gotchas
+
+- There have been some complications with stencil loading elements polyfills in Edge. So, some APIs that are available in Chrome and Firefox aren't available in Edge.
+- Internet Explorer is _not_ recommended to use with our web components.
+- We have a [wiki](https://github.com/Esri/calcite-app-components/wiki/Stencil-Tidbits#gotchas) of gotchas we've found with Stencil.
```

Comment:
I think we can revamp the wiki, move these bullet points there and and have this section link to it instead of having two gotchas sections.